### PR TITLE
added new query SQS policy with Public Access for ansible

### LINKED ISSUE #1588: Add SQS policy with Public Access query for Ansible
### Platform
Ansible

### Provider
AWS

### Description
SQS policy with public access

### PR DESCRIPTION
**Pull request by soniantao**
Closes #1588

### CODE REVIEW

File: assets/queries/ansible/aws/sqs_policy_with_public_access/query.rego [hunk 1]
```diff
@@ -0,0 +1,45 @@
+package Cx
+
+
+CxPolicy [ result ] {
+  document := input.document[i]
+  tasks := getTasks(document)
+  task := tasks[t]
+  sqsPolicy := task["community.aws.sqs_queue"]
+  sqsPolicyName := task.name
+  contains(sqsPolicy.policy.Statement[_].Principal,"*")
```
Comment (joaoReigota1) [inline assets/queries/ansible/aws/sqs_policy_with_public_access/query.rego:10]: We should check if the Effect is "Allow" and if the Action is "*" as well

File: assets/queries/ansible/aws/sqs_policy_with_public_access/query.rego [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+                "keyActualValue": 	sprintf("name=%s.{{community.aws.sqs_queue}}.policy.Principal is equal to '*'", [sqsPolicyName]),
+              }
+}
+CxPolicy [ result] {
+  document := input.document[i]
+  tasks := getTasks(document)
+  task := tasks[t]
+  sqsPolicy := task["community.aws.sqs_queue"]
+  sqsPolicyName := task.name
+  contains(sqsPolicy.policy.Statement[_].Principal[j].AWS,"*")
```
Comment (joaoReigota1) [inline assets/queries/ansible/aws/sqs_policy_with_public_access/query.rego:27]: Principal[j].AWS can also be an array we should check for Principal[j].AWS[_] as well

File: assets/queries/ansible/aws/sqs_policy_with_public_access/metadata.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+  "id": "SQS_policy_with_Public_Access",
+  "queryName": "SQS policy with Public Access",
+  "severity": "MEDIUM",
+  "category": null,
```
Comment (joaoReigota1) [inline assets/queries/ansible/aws/sqs_policy_with_public_access/metadata.json:5]: change the category to "Identity and Access Management"

**Review by joaoReigota1 — CHANGES_REQUESTED** [review]

**Comment by markmishaevcx** [comment]
Scan submitted to Checkmarx

**Comment by markmishaevcx** [comment]
Logo<br />**Checkmarx SAST - Scan Summary & Details**
### Cx-SAST Summary
Total of 4 vulnerabilities<br />High&nbsp;**0 High**<br />Medium&nbsp;**3 Medium**<br />Low&nbsp;**1 Low**<br />Info&nbsp;**0 Info**<br />
### Violation Summary
Medium&nbsp;**3** **Medium**<br />
[View more details on Checkmarx UI](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1049010&projectid=703)
### Cx-SAST Details
|Lines|Severity|Category|File|Link|
---|---|---|---|---
[30](https://github.com/Checkmarx/kics/blob/Feature/sqs_public/cmd/builder/main.go#L30) |Medium|Path_Traversal|cmd/builder/main.go|[Checkmarx](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1049010&projectid=703&pathid=3)
[83](https://github.com/Checkmarx/kics/blob/Feature/sqs_public/test/main_test.go#L83) |Medium|Denial_Of_Service_Resource_Exhaustion|test/main_test.go|[Checkmarx](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1049010&projectid=703&pathid=2)
[52](https://github.com/Checkmarx/kics/blob/Feature/sqs_public/pkg/kics/service.go#L52) |Medium|Denial_Of_Service_Resource_Exhaustion|pkg/kics/service.go|[Checkmarx](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1049010&projectid=703&pathid=1)

**Review by joaoReigota1 — APPROVED** [review]

**Pull request merged**